src/AsyncTCP.cpp

```diff
@@ -1383,10 +1425,17 @@ void AsyncServer::begin(){
     }
 
     ip_addr_t local_addr;
-    ip_addr_set_ip4_u32(&local_addr, _addr);
-/*    local_addr.type = bind_type;
+#if ESP_IDF_VERSION_MAJOR < 5
+    // ip_addr_set_ip4_u32(&local_addr, _addr);
+    local_addr.u_addr.ip4.addr = _addr;
+    local_addr.type = IPADDR_TYPE_V4;
+    ip_clear_no4(&local_addr);
+    /*    local_addr.type = bind_type;
     local_addr.u_addr.ip4.addr = (uint32_t) _addr;
     memcpy(local_addr.u_addr.ip6.addr, static_cast<const uint32_t*>(_addr6), sizeof(uint32_t) * 4); */
+#else
```

Comment:
Note to ESPHome devs: this part is weird and should consider IPv6. I've left it the way it was but I think there is a missing piece here in the case an `IPv6Address` has been passed to the constructor

Reply:
Many thanks! I just looked at the old code and `web_server` in ESPHome could never have worked with IPv6 on ESP32/Arduino. I was sure I tested it, but apparently not. I'm not sure I could get it to work on current Arduino version without breaking my back.

Reply:
I've reworked this part in my fork just this evening